File: apps/teacher_app/lib/src/features/home/domain/entities/news_item.dart
class NewsItem {
  final String id;
  final String title;
  final String content;
  final String date;
  final String imageUrl;

  NewsItem({
    required this.id,
    required this.title,
    required this.content,
    required this.date,
    required this.imageUrl,
  });

  factory NewsItem.fromJson(Map<String, dynamic> json) {
    return NewsItem(
      id: json['id'],
      title: json['title'],
      content: json['content'],
      date: json['date'],
      imageUrl: json['imageUrl'] ?? '',
    );
  }
}

File: backend/src/main/java/com/smartsyllabus/entity/NewsItem.java
package com.smartsyllabus.entity;

import java.util.Date;

public class NewsItem {
    private String id;
    private String title;
    private String content;
    private Date date;

    public NewsItem() {}

    public NewsItem(String id, String title, String content, Date date) {
        this.id = id;
        this.title = title;
        this.content = content;
        this.date = date;
    }

    public String getId() { return id; }
    public String getTitle() { return title; }
    public String getContent() { return content; }
    public Date getDate() { return date; }

    public void setId(String id) { this.id = id; }
    public void setTitle(String title) { this.title = title; }
    public void setContent(String content) { this.content = content; }
    public void setDate(Date date) { this.date = date; }
}